File: lib/models/muscle.dart
class Muscle {
  final int? id;
  final String? name;
  final String? image;
  final String? description;
  final DateTime? createdAt;
  final DateTime? updatedAt;
  final int activeStatus;

  Muscle({
    this.id,
    this.name,
    this.image,
    this.description,
    this.createdAt,
    this.updatedAt,
    this.activeStatus = 1,
  });

  factory Muscle.fromJson(Map<String, dynamic> json) {
    return Muscle(
      id: json['id'],
      name: json['name'],
      image: json['image'],
      description: json['description'],
      createdAt: json['created_at'] != null ? DateTime.parse(json['created_at']) : null,
      updatedAt: json['updated_at'] != null ? DateTime.parse(json['updated_at']) : null,
      activeStatus: json['active_status'],
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'id': id,
      'name': name,
      'image': image,
      'description': description,
      'created_at': createdAt?.toIso8601String(),
      'updated_at': updatedAt?.toIso8601String(),
      'active_status': activeStatus,
    };
  }
}

File: android/app/src/main/kotlin/com/example/yogi_application/model/Muscle.kt
package com.yogitech.yogi_application.model

import kotlinx.serialization.SerialName
import kotlinx.serialization.Serializable
import kotlinx.serialization.encodeToString
import kotlinx.serialization.json.Json


@Serializable
data class Muscle(
    val id: Int,
    val name: String,
    val image: String,
    val description: String,
    @SerialName("created_at") val createdAt: String,
        @SerialName("updated_at") val updatedAt: String,
        @SerialName("active_status") val activeStatus: Int
) {
    companion object {
        fun fromJson(source: String): Muscle {
            return Json.decodeFromString(source)
        }
    }

    fun toJson(): String {
        return Json.encodeToString(this)
    }
}
